src/tape_drivers/linux/lin_tape/lin_tape_ibmtape.c

```diff
@@ -2228,23 +2229,13 @@ int lin_tape_ibmtape_logsense_page(void *device, const uint8_t page, const uint8
 
 	if (rc != DEVICE_GOOD) {
 		lin_tape_ibmtape_process_errors(device, rc, msg, "logsense page", true);
-	}
-	else {
+	} else {
 		memcpy(buf, log_page.data, size);
 	}
 
-	return rc;
-}
-
-int lin_tape_ibmtape_logsense(void *device, const uint8_t page, unsigned char *buf, const size_t size)
-{
-	int ret = 0;
-	struct lin_tape_ibmtape *priv = (struct lin_tape_ibmtape *) device;
-
-	ltfs_profiler_add_entry(priv->profiler, NULL, TAPEBEND_REQ_ENTER(REQ_TC_LOGSENSE));
-	ret = lin_tape_ibmtape_logsense_page(device, page, 0, buf, size);
 	ltfs_profiler_add_entry(priv->profiler, NULL, TAPEBEND_REQ_EXIT(REQ_TC_LOGSENSE));
-	return ret;
+
+	return logpage.len;
```

Comment:
Before this change, the ret value of lin_tape_ibmtape_logsense() is actually the ret code of ioctl SIOC_LOG_SENSE10_PAGE. With this change, the ret value was changed to length of data in buffer, which dooms to fail the check below in [lin_tape_ibmtape_remaining_capacity()](https://github.com/LinearTapeFileSystem/ltfs/pull/287/commits/1fbde528e7aa0ce2a253fbd8511dd91064c66cd6#diff-72823ff6a06c061d93257e41fe1a0436acb899d0b7872f8d75372cd119f4f134R3389), even if the IOCTL is actually succeed.
We are hitting "LTFS30457I Cannot get remaining capacity: get log page 0x17 failed ..." error on an IBM tape autoloader. I think this is probably the root cause?